### ISSUE TITLE
Does not work in WebGL due to threading

### ISSUE THREAD

**Issue by isurfraz** [Gazuntype/GraphQL-Unity-Client#20]
Can a version using co-routines be added?

**Comment by zmays** [comment]
+1

**Comment by jtiret** [comment]
I have been able to make it run on a WebGL build. What I basically did was to rely on [UniTask](https://github.com/Cysharp/UniTask) instead of the AsyncAwaitUtil package integrated with the GraphQL client.

I didn't retry all the steps I followed on a fresh project so I may forget something but I'll do my best to be comprehensive:

1. Install the [UniTask package](https://github.com/Cysharp/UniTask). In my case, I did the install [using the Git URL](https://github.com/Cysharp/UniTask#install-via-git-url) but either way should be fine.
2. Open the [GraphQL client Unity package](https://github.com/Gazuntype/graphQL-client-unity#import-unitypackage) with Unity. 
3. When prompted to select the files you wish to import, uncheck the `AsyncAwaitUtil` folder under `plugins`.
4. Now that everything is imported, you should see the following compile error in `HttpHandler.cs` on line 33 and 58: 

> 'UnityWebRequestAsyncOperation' does not contain a definition for 'GetAwaiter' and no accessible extension method 'GetAwaiter' accepting a first argument of type 'UnityWebRequestAsyncOperation' could be found (are you missing a using directive or an assembly reference?)

5. That's because we didn't import the `AsyncAwaitUtil` which provides the definition for `GetAwaiter`. To fix this, import the interface provided by UniTask by adding this line at the top of `HttpHandler.cs`: `using Cysharp.Threading.Tasks;` as follows:

<img width="321" alt="Screenshot_2020-12-08_at_10_07_29" src="https://user-images.githubusercontent.com/7695933/101463218-7265e200-393d-11eb-9f39-9eed8cd0ead9.png">

Everything should work fine now.

Unfortunately, the steps above imply to update the code of a third-party package (in that case, `graphQL-client-unity`), which is never a good idea as this will break upon package updates. It would be better if we added this directly in the package itself, but there might be unwanted side effects by relying on UniTask instead of the provided package. 
Maybe @Gazuntype could give his opinion on this :)

**Comment by cmadh** [comment]
I guess subscriptions will not work either because websocket-connections in webgl are only possible via a seperate jslib?

**Comment by jtiret** [comment]
@cmadh that's right, the implementation of WebSocket in this library doesn't work with WebGL. I had to do it myself. In case you want to use WebSocket with WebGL, I advise you to use [this library](https://github.com/endel/NativeWebSocket). Even for uses cases without WebGL as a build target, I find its APIs easier to use than the regular `System.Net.WebSockets`.

**Comment by cmadh** [comment]
> @cmadh that's right, the implementation of WebSocket in this library doesn't work with WebGL. I had to do it myself. In case you want to use WebSocket with WebGL, I advise you to use [this library](https://github.com/endel/NativeWebSocket). Even for uses cases without WebGL as a build target, I find its APIs easier to use than the regular `System.Net.WebSockets`.

@jtiret thank you! Any chance you make your modified fork public? Do you know if the ws-library you recommended is able to handle multiple websocket-instances? I have multiple plugins relying on a websocket-connection and so far I havn't found a solution to use them in parallel because all the alternative webgl-ws-libs only work with a single instance...

**Comment by memark** [comment]
> When prompted to select the files you wish to import, uncheck the AsyncAwaitUtil folder under plugins.

@jtiret I'm afraid the `plugins` folder doesn't even show up when importing this package:
https://github.com/gazuntype/graphQL-client-unity/blob/master/UnityPackages/graphQl-client-unity-v2.unitypackage

Or are there different ways of importing? 

![image](https://user-images.githubusercontent.com/318504/132803984-0c56eeb0-7322-48ae-916d-8c049668b797.png)
